packages/vue/src/components/reset-password.vue

```diff
@@ -0,0 +1,148 @@
+        data-amplify-authenticator-resetpassword
+        @submit.prevent="onResetPasswordSubmit"
+        @change="onChange"
+      >
+        <base-heading>
+          {{ resetPasswordHeading }}
+        </base-heading>
+        <base-field-set :disabled="actorState.matches('resetPassword.pending')">
+          <base-label data-amplify-resetpassword-label>
+            <base-text>Username</base-text>
+            <base-input
+              name="username"
+              placeholder="Enter your username"
+              autocomplete="username"
+              required
+              type="username"
+            ></base-input>
+          </base-label>
+        </base-field-set>
+
+        <base-footer>
+          <template #footert="{ slotData }">
+            <slot
+              name="footer"
+              :info="slotData"
+              :onBackToSignInClicked="onBackToSignInClicked"
+              :onResetPasswordSubmit="onResetPasswordSubmit"
+            >
+            </slot>
+          </template>
+          <base-button type="button" @click.prevent="onBackToSignInClicked">
+            {{ backSignInText }}</base-button
+          >
+          <base-spacer />
+          <base-button
+            :disabled="actorState.matches('resetPassword.pending')"
+            >{{ resetPasswordText }}</base-button
+          >
+        </base-footer>
+        <base-box data-ui-error>
+          {{ actorState?.context?.remoteError }}
+        </base-box>
+      </base-form>
+    </base-wrapper>
+  </slot>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed, ComputedRef } from 'vue';
+import BaseHeading from './primitives/base-heading.vue';
+import BaseFieldSet from './primitives/base-field-set.vue';
+import BaseLabel from './primitives/base-label.vue';
+import BaseSpacer from './primitives/base-spacer.vue';
+import BaseButton from './primitives/base-button.vue';
+import BaseFooter from './primitives/base-footer.vue';
+import BaseText from './primitives/base-text.vue';
+import BaseInput from './primitives/base-input.vue';
+import BaseForm from './primitives/base-form.vue';
+import BaseBox from './primitives/base-box.vue';
+import BaseWrapper from './primitives/base-wrapper.vue';
+
+import { useAuth } from '../composables/useAuth';
+
+import {
+  BACK_SIGN_IN_TEXT,
+  RESET_PASSWORD_HEADING,
+  RESET_PASSWORD_TEXT,
+} from '../defaults/DefaultTexts';
+import { ResetPasswordSetupReturnTypes, SetupEventContext } from '../types';
+import { getActorState, SignInState } from '@aws-amplify/ui-core';
+
+export default defineComponent({
+  components: {
+    BaseBox,
+    BaseHeading,
+    BaseFieldSet,
+    BaseForm,
+    BaseLabel,
+    BaseSpacer,
+    BaseButton,
+    BaseFooter,
+    BaseText,
+    BaseInput,
+    BaseWrapper,
+  },
+  inheritAttrs: false,
+  setup(_, { emit, attrs }: SetupEventContext): ResetPasswordSetupReturnTypes {
+    const { state, send } = useAuth();
+    const actorState: ComputedRef<SignInState> = computed(() =>
```

Comment:
```suggestion
    const actorState: ComputedRef<ResetPasswordState> = computed(() =>
```

Reply:
👍

Reply:
FYI, I had to add an `as ComputedRef<ResetPasswordState>` for my VSCode to see it correctly. But that should be fine